designs/extensible-cli-and-scaffolding-plugins-phase-1-5.md

```diff
@@ -46,36 +46,42 @@ Plugin chaining solves the aforementioned problems but the current plugin API, a
 
 Design a Plugin API that combines the current [`Subcommand`](../pkg/plugin/interfaces.go) and
 [`RunOptions`](../pkg/plugins/internal/cmdutil/cmdutil.go) interfaces and enables plugin-chaining.
```

Comment:
Inside of the Motivation, we can add some clarifications such as;

### Goals 

- Allow initing a project with specific optional plugins behaviours by default
- Allow using optional plugins in subcommands to address specific needs and helpers its users
- Allow the plugin's information to be stored in order to provide tooling, helpers and smarter custom and optional plugins which will be able to do the scaffolds accordingly with the current configuration of the project to check their capabilities
- Bring the facilities discussed for the preliminary plugin design ( // todo add the link for 1.0 plugin version) introduced in Kubebuilder which allows more re-usage of the code and will allow further solutions to improve their maintainability. For more info; add the plugin 2.0 issue 
- Allow the preliminary implemetation for we start to composing plugins to address requirements such as to provide Hybrid operators whcih would be a composition of one or keep easier consumers provide and maintain many languages operator plugins where would be possible re-use and centralized the common needs between all types. more info https://github.com/operator-framework/operator-sdk/issues/670 and https://github.com/operator-framework/operator-sdk/issues/2745

Reply:
Added the non-goal and a paragraph describing the plugin override per-command.